server/controllers/parties.js

```diff
@@ -1,97 +1,118 @@
-      ],
     });
   }
 
   // READ - Get All Parties
   static getAll(req, res) {
-    return res.status(200).json({
-      status: 200,
-      data: partiesDb,
+    db.query('SELECT * FROM parties ORDER BY id ASC', (err, foundParties) => {
+      // callback
+      if (err) {
+        res.status(400).json({
+          status: 400,
+          error: err.stack,
+        });
+      }
+      return res.status(200).send({
+        status: 200,
+        data: foundParties.rows,
+      });
     });
   }
 
   // READ - Get a particular party
   static getOne(req, res) {
+    const text = 'SELECT * FROM parties WHERE id = $1';
     const id = parseInt(req.params.id, 10);
-    const selectedParty = partiesDb.find(party => party.id === id);
-    if (!selectedParty) {
-      return res.status(404).json({
+    db.query(text, [id], (err, foundParty) => {
+      if (err) {
+        res.status(400).json({
+          status: 400,
+          error: err.stack,
+        });
+      }
+      if (foundParty.rowCount !== 0) {
+        return res.status(200).json({
+          status: 200,
+          data: [foundParty.rows[0]],
+        });
+      }
+      return res.status(404).send({
         status: '404',
-        error: 'The party was not found',
+        error: 'The office was not found',
       });
-    }
-    return res.status(200).json({
-      status: 200,
-      data: [selectedParty],
     });
   }
 
   // PUT - Update Party
   static update(req, res) {
     const id = parseInt(req.params.id, 10);
-    const partyIndex = partiesDb.findIndex(party => party.id === id);
-
-    if (partyIndex === -1) {
-      return res.status(404).json({
+    const { name } = req.body;
+    db.query('UPDATE parties SET name = $1 WHERE id = $2 RETURNING *', [name, id], (err, updatedParty) => {
+      if (err) {
+        res.status(400).send({
+          status: 400,
+          error: err.stack,
+        });
+      }
+      if (updatedParty.rowCount !== 0) {
+        return res.status(200).json({
+          status: 200,
+          data: [updatedParty.rows[0]],
+        });
+      }
+      return res.status(404).send({
         status: '404',
         error: 'The party was not found',
       });
-    }
-    partiesDb[partyIndex].name = req.body.name;
-    return res.status(200).json({
-      status: '200',
-      data: [partiesDb[partyIndex]],
     });
   }
 
   // DELETE - Delete Party
   static delete(req, res) {
     const id = parseInt(req.params.id, 10);
-    const partyIndex = partiesDb.findIndex(party => party.id === id);
-
-    if (partyIndex === -1) {
-      return res.status(404).json({
+    db.query('DELETE FROM parties WHERE id = $1 RETURNING *', [id], (err, response) => {
+      if (err) {
+        res.status(400).send({
+          status: 400,
+          error: err.stack,
+        });
+      }
+      if (response.rowCount !== 0) {
+        return res.status(200).json({
+          status: 200,
+          data: [{ message: `User Successfully deleted with ID: ${id}` }],
```

Comment:
You don't need to expose the id of the deleted user. The message `User Successfully deleted ` is enough.

Reply:
Ok